### PR TITLE
memory_read and memory_write

### PR DESCRIPTION
**Pull request by murduk**
#### What is wrong?
Currently, the Computation object houses lots of information and sub-APIs for things like maniputlating the stack, consuming gas, etc. This creates a lot of public API surface area since we are effectively support two public APIs for anything needed from a sub-property of the Computation object.

Computation.memory is the API for getting at the Memory
Computation.memory.read is the API for reading from memory.

#### How can it be fixed
Computation.memory -> Computation._memory
Computation.memory.read() -> Computation.memory_read()
Computation.memory.write() -> Computation.memory_write()

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: evm/computation.py [hunk 1]
```diff
@@ -133,6 +133,10 @@ def should_return_gas(self):
     def should_erase_return_data(self):
         return self.is_error and self._error.erases_return_data
 
+    @property
+    def memory_bytes(self):
```
Comment (pipermerriam) [inline evm/computation.py]: We've tended to stay away from APIs that are only used in testing.  What do you think about just reaching down into `computation._memory.bytes` in the tests and removing this property?

Comment (murduk) [reply evm/computation.py]: Sure, I wasn't very comfortable to modify all the tests .

**Comment by murduk** [comment]
@pipermerriam hello, I am not super familiar with github. Do I have to do anything now? Thank you

**Comment by murduk** [comment]
@pipermerriam sorry, I get it now, builds failed. I will fix right away.

**Comment by pipermerriam** [comment]
> I am not super familiar with github. Do I have to do anything now?

Not a problem.  Everyone is new at some point.

Once you open a pull request, all you need to do is continue pushing your changes to your branch and the PR will update itself, trigger new test runs, etc.  When your tests are green/passing please ping me again and I'll give this another review.

Thanks for your contribution!

**Comment by murduk** [comment]
@pipermerriam so sorry for bothering, could you help on fixing the first step of the build? The rest builds without an error. Thank you

**Comment by pipermerriam** [comment]
@murduk see: https://travis-ci.org/ethereum/py-evm/jobs/354548922#L506-L507

You have two minor linting errors that need to be cleaned up.  You should be able to see these locally by running `flake8 evm`.

**Pull request merged**